Mathlib/AlgebraicTopology/Quasicategory/StrictBicategory.lean

```diff
@@ -0,0 +1,71 @@
+/-
+Copyright (c) 2025 Emily Riehl. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Emily Riehl
+-/
+
+import Mathlib.CategoryTheory.Bicategory.CatEnriched
+import Mathlib.AlgebraicTopology.Quasicategory.Basic
+import Mathlib.AlgebraicTopology.SimplicialCategory.Basic
+import Mathlib.AlgebraicTopology.SimplicialSet.NerveAdjunction
+
+/-!
+# The strict bicategory of quasicategories
+
+In this file we define a strict bicategory whose objects are quasicategories,
+whose `1`-morphisms are maps of simplicial sets, and whose `2`-morphisms are homotopy classes of
+homotopies between them, defined using the simplicial interval `Δ[1]`.
+
+This is defined by transfering the simplicial enrichment of the category of quasicategories to
+a categorical enrichment by applying `hoFunctor : SSet ⥤ Cat` to the hom-objects.
+
+This strict bicategory serves as a setting to develop the formal category theory of quasicategories.
+
+## References
+
+* [Emily Riehl and Dominic Verity, Elements of ∞-Category Theory][RiehlVerity2022]
+* [Emily Riehl and Dominic Verity, The 2-category theory of quasi-categories][RiehlVerity2015]
+
+-/
+
+universe u
+
+namespace SSet
+
+open CategoryTheory Simplicial
+
+/-- `QCat` is the category of quasi-categories defined as the full subcategory of the category
+`SSet` of simplicial sets. -/
+abbrev QCat := ObjectProperty.FullSubcategory Quasicategory
+
+/-- `QCat` is the category of quasi-categories defined as the full subcategory of the category
+`SSet` of simplicial sets. -/
+instance : Category QCat := ObjectProperty.FullSubcategory.category Quasicategory
+
+/-- As a full subcategory of `SSet`, the category `QCat` is a simplicially enriched ordinary
+category. -/
+instance QCat.SimplicialCat : SimplicialCategory QCat where
+  Hom X Y := X.obj.functorHom Y.obj
+  id X := Functor.natTransEquiv.symm (𝟙 X.obj)
+  comp X Y Z := { app := fun _ ⟨f, g⟩ => f.comp g }
+  homEquiv := Functor.natTransEquiv.symm
+
+/-- `QCat` obtains a `Cat`-enriched ordinary category structure by applying `hoFunctor` to the
+hom objects in its `SSet`-enriched ordinary structure. -/
+noncomputable instance QCat.CatEnrichedOrdinaryCat : EnrichedOrdinaryCategory Cat QCat :=
+  TransportEnrichment.enrichedOrdinaryCategory QCat hoFunctor hoFunctor_pro_normal_monoidal
```

Comment:
This *may* be made computable by changing the `Function.Bijective` assumption in `TransportEnrichment.enrichedOrdinaryCategory` to the data of an `equiv`, assuming it is easy to replace the lemma `hoFunctor_pro_normal_monoidal` by the definition of an `equiv` (which seems reasonable).

Reply:
Ah, another issue is that the oplax-monoidal instance for `hoFunctor` is probably computable (for free), but not the lax-monoidal structure.

Reply:
I may have a completely different approach for proving that `hoFunctor` is monoidal, by defining a computable inverse `X.HomotopyCategory × Y.HomotopyCategory) ⥤ (X ⊗ Y).HomotopyCategory`:
https://github.com/joelriou/mathlib4/blob/2b3cf3027e75ed4f5d3735a8b564d437d0b5f170/Mathlib/AlgebraicTopology/SimplicialSet/HoFunctorMonoidal.lean#L135